aiida/transports/plugins/local.py

```diff
@@ -759,27 +759,34 @@ def _exec_command_internal(self, command, **kwargs):  # pylint: disable=unused-a
 
         return proc.stdin, proc.stdout, proc.stderr, proc
 
-    def exec_command_wait(self, command, **kwargs):
+    def exec_command_wait_bytes(self, command, stdin=None, **kwargs):
         """
         Executes the specified command and waits for it to finish.
 
         :param command: the command to execute
 
         :return: a tuple with (return_value, stdout, stderr) where stdout and
-                 stderr are strings.
+                 stderr are bytes.
         """
-        stdin = kwargs.get('stdin')
         local_stdin, _, _, local_proc = self._exec_command_internal(command)
 
         if stdin is not None:
+            # Implicitly assume that the desired encoding is 'utf-8' if I receive a string.
+            # Also, if I get a StringIO, I jsut read it all in memory and put it into a BytesIO.
+            # Clearly not memory effective - in this case do not use a StringIO, but pass directly a BytesIO
+            # that will be read line by line, if you have a huge stdin and care about memory usage.
             if isinstance(stdin, str):
-                filelike_stdin = io.StringIO(stdin)
+                filelike_stdin = io.BytesIO(stdin.encode('utf-8'))
+            elif isinstance(stdin, bytes):
+                filelike_stdin = io.BytesIO(stdin)
+            elif isinstance(stdin, io.TextIOBase):
+                filelike_stdin = io.BytesIO(stdin.read().encode('utf-8'))
             else:
                 filelike_stdin = stdin
 
             try:
                 for line in filelike_stdin.readlines():
-                    local_stdin.write(line.encode('utf-8'))  # the Popen.stdin/out/err are byte streams
+                    local_stdin.write(line)  # the Popen.stdin/out/err are byte streams
```

Comment:
The `for line in filelike_stdin.readlines():` should probably be a `for line filelike_stdin:` to avoid another copy of the content as list of strings.

Reply:
Thanks!